engine/src/ast/function_expr.rs

```diff
@@ -11,24 +11,21 @@ use serde::Serialize;
 #[derive(Debug, PartialEq, Eq, Clone, Serialize)]
 #[serde(tag = "kind", content = "value")]
 pub(crate) enum FunctionCallArgExpr<'s> {
-    LhsFieldExpr(LhsFieldExpr<'s>),
+    IndexExpr(IndexExpr<'s>),
     Literal(RhsValue),
 }
 
 impl<'s> FunctionCallArgExpr<'s> {
-    pub fn uses(&self, field: Field<'s>) -> bool {
+    pub fn uses(&self, field: &Field<'s>) -> bool {
         match self {
-            FunctionCallArgExpr::LhsFieldExpr(lhs) => lhs.uses(field),
+            FunctionCallArgExpr::IndexExpr(index) => index.uses(field),
```

Comment:
I'd use the variable name `expr` here, `index` is a little misleading as it confuses me with the index from `FieldIndex`.